package/kernel/mac80211/files/lib/wifi/mac80211.sh

```diff
@@ -76,21 +76,27 @@ detect_mac80211() {
 		config_foreach check_mac80211_device wifi-device
 		[ "$found" -gt 0 ] && continue
 
-		mode_band="g"
-		channel="11"
-		htmode=""
-		ht_capab=""
-
-		iw phy "$dev" info | grep -q 'Capabilities:' && htmode=HT20
+		if ! [ iw phy "$dev" info | grep -q '58320 MHz' ]; then
+			mode_band="ad"
+			channel="1"
+			htmode=""
+			ht_capab=""
+		else
+			mode_band="g"
+			channel="11"
+			htmode=""
+			ht_capab=""
 
-		iw phy "$dev" info | grep -q '5180 MHz' && {
```

Comment:
I'd personally prefer if/elif logic to the nested approach here. It won't be much difference in the end, but I think it would be more obvious to the reader. Also moved around definitions a bit:
```
htmode=""
ht_capab=""

if iw phy "$dev" info | grep -q '58320 MHz'; then
	mode_band="ad"
	channel="1"
elif iw phy "$dev" info | grep -q '5180 MHz'; then
	mode_band="a"
	channel="36"
	iw phy "$dev" info | grep -q 'Capabilities:' && htmode=HT20
	iw phy "$dev" info | grep -q 'VHT Capabilities' && htmode="VHT80"
else
	mode_band="g"
	channel="11"
	iw phy "$dev" info | grep -q 'Capabilities:' && htmode=HT20
fi

[ -n "$htmode" ] && ht_capab="set wireless.radio${devidx}.htmode=$htmode"
```